layer/layer.go

```diff
@@ -137,6 +138,22 @@ type RWLayer interface {
 	ApplyDiff(diff io.Reader) (int64, error)
 }
 
+// DirectMountRWLayer defines a layer that can return the underlying
+// mounts to be performed (as opposed to the default Mount method that
+// does the mount and then return a ContainerFS).
+// It's currently only intended to be used with the BuildKit adapter to
+// enable optimizations that require knowing the actual underlying mount.
+type DirectMountRWLayer interface {
```

Comment:
Wondering if this interface should be defined on the receiver end (in builder-next/adapters/snapshotter) 🤔